qiskit_dynamics/pulse/pulse_simulator.py

```diff
@@ -0,0 +1,330 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import time
+import datetime
+import uuid
+
+from typing import Dict, Iterable, List, Optional, Union
+import numpy as np
+
+from qiskit import pulse
+from qiskit.pulse import Schedule, ScheduleBlock
+from qiskit.pulse.channels import AcquireChannel, DriveChannel, MeasureChannel, ControlChannel
+from qiskit.pulse.transforms.canonicalization import block_to_schedule
+from qiskit.providers.backend import BackendV2
+from qiskit.result import Result
+
+from qiskit import QiskitError
+from qiskit.quantum_info import Statevector, DensityMatrix
+
+from qiskit_dynamics import Solver
+from qiskit_dynamics.array import Array
+from qiskit_dynamics.models import HamiltonianModel
+
+from .dynamics_job import DynamicsJob
+from .pulse_utils import _get_dressed_state_decomposition, _get_lab_frame_static_hamiltonian
+
+
+class PulseSimulator(BackendV2):
+    def __init__(
+        self,
+        solver: Solver,
+        subsystem_dims,
+        subsystem_labels: Optional[List[int]] = None,
+        name: Optional[str] = 'PulseSimulator',
+    ):
```

Comment:
In https://github.com/Qiskit/qiskit-experiments/pull/925 we went for a model centric approach. I guess that all of those end-up going into a solver anyhow but we do need to know things like how many levels we are dealing with. 

How do you plan on giving a discriminator to the backend? Not needed now but you could also think of the kernel. The question is in fact more general: How to implement the readout chain and which arguments does it need? The first version would at least need IQ data and counts, both single-shot and average.

In https://github.com/Qiskit/qiskit-experiments/pull/925 we either go from state vector to counts (Level 2) or we go from state vector to IQ data which we sample from a predetermined distribution. One thing that this is still lacking is a discriminator to go from IQ data to counts which would allow us to capture mis-classification of, e.g., the two state as the 1 state.

Reply:
I think all these things are just internal configuration that would be handled through setting relevant simulator options with `set_options`

Reply:
Agree with Chris. Adding IQ data generation will be left for a subsequent PR, and enabling any user configuration of this process can be handled by designing/implementing options.